### ISSUE TITLE
[Feature] Use GoLang 1.20 to fix DNS resolution on macOS

### ISSUE THREAD

**Issue by tristanmorgan** [hashicorp/consul#16265]
#### Feature Description

Related to  [#3267](https://github.com/hashicorp/consul/issues/3267), when Consul is compiled with golang on 1.20 the updated resolving logic for macOS allows the use of scoped queries.

Example output of release build
```bash
$ export CONSUL_HTTP_ADDR=consul.service.consul
$ consul version
Consul v1.14.4
Revision dae670fe
Build Date 2023-01-26T15:47:10Z
Protocol 2 spoken by default, understands 2 to 3 (agent will automatically use protocol >2 when speaking to compatible agents)
$ consul members
Error retrieving members: Get "http://consul.service.consul/v1/agent/members?segment=_all": dial tcp: lookup consul.service.consul on 10.0.0.1:53: no such host
```

Example output of GoLang 1.20 build with CGO_ENABLED=0 and GOOS=darwin inside golang:alpine docker container
```bash
$ export CONSUL_HTTP_ADDR=consul.service.consul
$ ./consul version
Consul v1.14.4
Build Date 1970-01-01T00:00:01Z
Protocol 2 spoken by default, understands 2 to 3 (agent will automatically use protocol >2 when speaking to compatible agents)
$ ./consul members
Node      Address            Status  Type    Build   Protocol  DC    Partition  Segment
myserver  10.0.0.23:8301     alive   server  1.14.4  3         dc1   default    <all>
```

#### Use Case(s)

To alow the use of scoped queries as described at [Forward DNS for Consul Service Discovery](https://developer.hashicorp.com/consul/tutorials/networking/dns-forwarding#macos-setup)

**Comment by tristanmorgan** [comment]
Fixed in #16263

**Comment by david-yu** [comment]
Thanks @tristanmorgan that PR you linked has been merged and will go out first in 1.15.0 later this month, and will be back ported to 1.14.x a few weeks afterwards next month.
